control-plane/plugin/src/resources/pool.rs

```diff
@@ -99,3 +144,26 @@ impl Get for Pool {
         Ok(())
     }
 }
+
+/// Check if the labels match the pool labels.
+pub(crate) fn labels_matched(
+    pool_labels: &HashMap<String, String>,
+    labels: &Option<String>,
+) -> Result<bool, Error> {
+    match labels {
+        Some(filter_labels) => {
+            for label in filter_labels.split(',') {
+                let [key, value] = label.split('=').collect::<Vec<_>>()[..] else {
```

Comment:
Seems we're only supporting '='
Can we update this to include '==' and '!=' ?

Reply:
This parsing will require regex. IMHO we can keep it simple in this PR and later take it as an enhancement if needed.

Reply:
Sounds good, but please then update doc comment to reflect this (ie remove the '==' and '!=' from the doc comment)

Reply:
Why do we need the `==`? What would that query look like? 
In k8s its equality and set based. I think if we want an enhancement we should go for the k8s way.
ref: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#list-and-watch-filtering

Reply:
kubectl get pods --help will show you the below option 
```
  -l, --selector='':
	Selector (label query) to filter on, supports '=', '==', and '!='.(e.g. -l key1=value1,key2=value2). Matching
	objects must satisfy all of the specified label constraints.
``` 

Refer https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#equality-based-requirement